### PR TITLE
chore(multi-env): add schema of the new env config file

### PR DESCRIPTION
**Pull request by qinezh**
1. add the schema of the new env config file (`config.dev.json`): `envConfig.json` according our previous discussion.
2. the type of `envConfig.ts` will be generated automatically after npm build.

### CODE REVIEW

File: packages/api/src/schemas/envConfig.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "$id": "https://github.com/OfficeDev/TeamsFx/schemas/v1.0/envConfig.json",
```
Comment (a1exwang) [inline packages/api/src/schemas/envConfig.json:3]: Should this "$id" be a valid URL? like `https://raw.githubusercontent.com/OfficeDev/TeamsFx/main/packages/api/src/schemas/envConfig.json

File: packages/api/src/schemas/envConfig.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      "properties": {
+        "subscriptionId": {
+          "type": "string",
+          "description": "The default subscription to provision Azure resources."
+        },
+        "resourceGroupName": {
+          "type": "string",
+          "description": "The default resource group of Azure resources."
+        },
+        "region": {
```
Comment (a1exwang) [inline packages/api/src/schemas/envConfig.json:22]: Should we be consistent with Azure's wording? 
image

Comment (qinezh) [reply packages/api/src/schemas/envConfig.json]: updated to use `location` for resource group to keep consistent with the ARM template schema.

File: packages/api/src/schemas/envConfig.json (hunk 1 above)
Comment (qinezh) [reply packages/api/src/schemas/envConfig.json]: Removed it to avoid confusion and we can add it back when it's needed.

**Review by a1exwang — APPROVED** [review]

File: packages/api/package-lock.json [hunk 3]
```diff
@@ -1257,7 +1326,7 @@
     },
     "asynckit": {
       "version": "0.4.0",
-      "resolved": "https://registry.npmjs.org/asynckit/-/asynckit-0.4.0.tgz",
+      "resolved": "https://msazure.pkgs.visualstudio.com/9660fff2-2363-48b0-9e15-64df2283e932/_packaging/MODS-NPM-Registry/npm/registry/asynckit/-/asynckit-0.4.0.tgz",
```
Comment (dooriya) [inline packages/api/package-lock.json:1329]: "MODS"? I guess you don't want to check in this.

File: packages/api/src/schemas/envConfig.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+   */
+  manifest: {
+    description?: string;
+    /**
+     * Configs to customize the Teams app manifest.
+     */
+    values: {
+      [k: string]: unknown;
+    };
+    [k: string]: unknown;
```
Comment (dooriya) [inline packages/api/src/schemas/envConfig.ts:45]: The `menifest` config interface is not finalized yet, should we include this in the schema now?

File: packages/api/package-lock.json (hunk 3 above)
Comment (qinezh) [reply packages/api/package-lock.json:1329]: Thanks, I'll fix it.

File: packages/api/src/schemas/envConfig.ts (hunk 4 above)
Comment (qinezh) [reply packages/api/src/schemas/envConfig.ts:45]: This is not for the finalized version but the M2's goal, it's open to be modified.

**Review by dooriya — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by 1yefuwang1 — COMMENTED** [review]
Will the schema file be scaffolded into developers' project?

**Comment by qinezh** [comment]
> Will the schema file be scaffolded into developers' project?

no, it won't.

**Review by 1yefuwang1 — APPROVED** [review]

**Pull request merged**